sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -278,4 +278,15 @@ object ResolveHints {
         h.child
     }
   }
+
+  /**
+   * Removes all the hints when `spark.sql.ignorePlanHints` is set.
+   * This is executed at the very beginning of the Analyzer to disable
+   * the hint functionality.
+   */
+  class DisableHints(conf: SQLConf) extends RemoveAllHints(conf: SQLConf) {
+    override def apply(plan: LogicalPlan): LogicalPlan = {
+      if (conf.getConf(SQLConf.IGNORE_HINTS)) super.apply(plan) else plan
```

Comment:
ignore or disable? we should be consistent

Reply:
+1. You are right. I will change the config to `spark.sql.optimizer.disableHints` to make it consistent. But could you please help answer @dongjoon-hyun's question on if we need a ".enable" at the end ?

Reply:
Hi, @dilipbiswal . It seems that he also meant`SQLConf.IGNORE_HINTS` should be `SQLConf.DISABLE_HINTS`.